server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNodes.java

```diff
@@ -161,6 +200,17 @@ public List<DesiredNode> nodes() {
         return List.copyOf(nodes.values());
     }
 
+    public List<DesiredNode> members() {
+        assert members.stream().allMatch(DesiredNode::isMember);
+        return members;
+    }
+
+    public List<DesiredNode> notMembers() {
```

Comment:
Like for the enum, I wonder if `newDesiredNodes` is better? I am not truly happy about that either though.